src/main/groovy/com/github/benmanes/gradle/versions/reporter/HtmlReporter.groovy

```diff
@@ -0,0 +1,287 @@
+package com.github.benmanes.gradle.versions.reporter
+
+import com.github.benmanes.gradle.versions.reporter.result.DependenciesGroup
+import com.github.benmanes.gradle.versions.reporter.result.Dependency
+import com.github.benmanes.gradle.versions.reporter.result.DependencyLatest
+import com.github.benmanes.gradle.versions.reporter.result.DependencyOutdated
+import com.github.benmanes.gradle.versions.reporter.result.DependencyUnresolved
+import com.github.benmanes.gradle.versions.reporter.result.Result
+import com.github.benmanes.gradle.versions.reporter.result.VersionAvailable
+import groovy.text.markup.MarkupTemplateEngine
+import groovy.transform.CompileStatic
+import groovy.transform.TupleConstructor
+import groovy.xml.MarkupBuilder
+
+import static com.github.benmanes.gradle.versions.updates.gradle.GradleReleaseChannel.CURRENT
+import static com.github.benmanes.gradle.versions.updates.gradle.GradleReleaseChannel.NIGHTLY
+import static com.github.benmanes.gradle.versions.updates.gradle.GradleReleaseChannel.NIGHTLY
+import static com.github.benmanes.gradle.versions.updates.gradle.GradleReleaseChannel.NIGHTLY
+import static com.github.benmanes.gradle.versions.updates.gradle.GradleReleaseChannel.NIGHTLY
+import static com.github.benmanes.gradle.versions.updates.gradle.GradleReleaseChannel.NIGHTLY
+import static com.github.benmanes.gradle.versions.updates.gradle.GradleReleaseChannel.RELEASE_CANDIDATE
+import static com.github.benmanes.gradle.versions.updates.gradle.GradleReleaseChannel.RELEASE_CANDIDATE
+import static com.github.benmanes.gradle.versions.updates.gradle.GradleReleaseChannel.RELEASE_CANDIDATE
+
+/**
+ * A xml reporter for the dependency updates results.
+ */
+@CompileStatic
+@TupleConstructor(callSuper = true, includeSuperProperties = true, includeSuperFields = true)
+class HtmlReporter extends AbstractReporter {
+
+  String header = """
```

Comment:
You may (or may not) prefer using [MarkupBuilder](https://mrhaki.blogspot.com/2009/10/groovy-goodness-creating-xml-with.html) instead of raw strings. I'm not much of a Groovy expert so lack an opinion and am happy either way.

Reply:
That was my first day :) went with mimicking the structure of the text report for consistency.